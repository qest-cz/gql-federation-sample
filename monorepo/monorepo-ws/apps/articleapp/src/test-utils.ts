import { testServer } from '@monorepo-ws/server';
import { gql } from 'apollo-server';
import { modules } from './resolvers/modules';
import { GqlArticle, GqlMutationCreateArticleArgs, GqlUser } from './resolvers/interfaces';
import {DataSource as ApolloDataSources} from 'apollo-datasource'
import { IArticleDataSource } from './data-sources/interfaces';
import { DataSources } from './main'; 
import { Article } from '@monorepo-ws/prisma-article-app-client';
import { exportArticle, exportArticles } from './services/article-services';

const testModules = [
  {
    typeDefs: gql`
      extend type Query {
        getUser: User!
      }
    `,
    resolvers: {
      Query: {
        getUser: (): Pick<GqlUser, 'id'> => {
          return { id: 1 };
        },
      },
    },
  },
];

const testArticles: Article[] = [{id: 1, title: "first title", authorId: 1},
                                {id: 2, title: "second title", authorId: 1}]

export const server = testServer([...modules, ...testModules], (): DataSources => ({article: new FakePrismaArticleDataSource()}));

class FakePrismaArticleDataSource extends ApolloDataSources implements IArticleDataSource{
    constructor(){
      super()
    }

    async getAllArticles(): Promise<GqlArticle[]>{
        return exportArticles(testArticles)
    }

    async createArticle(newArticle: GqlMutationCreateArticleArgs): Promise<GqlArticle>{
        const article: Article = {id: 3, title: 'added title', authorId: 1} 
        return exportArticle(article)
    }

    async GetArticleByAuthorId(id: number) : Promise<GqlArticle[]>{
      const articles = testArticles.filter(x => x.authorId == id)
      return exportArticles(articles)
    }
}
