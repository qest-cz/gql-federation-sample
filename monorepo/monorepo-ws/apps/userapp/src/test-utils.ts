import { gql } from 'apollo-server';
import { testServer} from '@monorepo-ws/server'; 
import { modules } from './resolvers/modules';

export const server = testServer(modules)
