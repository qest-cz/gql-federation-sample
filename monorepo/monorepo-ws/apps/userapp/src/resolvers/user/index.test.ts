import { gql } from 'apollo-server';
import { server as testServer } from '../../test-utils';
import { getUsers } from './index';

describe('test users', () => {
  test('count array is 4', () => {
    const len: number = getUsers().length;
    expect(len).toBe(4);
  });
});

const query = gql`
  query {
    users {
      id
      name
      married
      age
      friends {
        name
      }
    }
  }
`;
const userByNameQuery = gql`
  query abc($name: String!) {
    getUserByName(name: $name) {
      id
      name
      married
      age
    }
  }
`;

describe('user resolver', () => {
  it('should return users', async () => {
    const result = await testServer.executeOperation({
      query,
    });
    expect(result.data).toMatchSnapshot({
      users: [
        {
          id: expect.any(String),
          name: expect.any(String),
          married: expect.any(Boolean),
          age: expect.any(Number),
          friends: [
            {
              name: expect.any(String),
            },
            {
              name: expect.any(String),
            },
          ],
        },
        {
          id: expect.any(String),
          name: expect.any(String),
          married: expect.any(Boolean),
          age: expect.any(Number),
          friends: [],
        },
        {
          id: expect.any(String),
          name: expect.any(String),
          married: expect.any(Boolean),
          age: expect.any(Number),
          friends: [],
        },
        {
          id: expect.any(String),
          name: expect.any(String),
          married: expect.any(Boolean),
          age: expect.any(Number),
          friends: [],
        },
      ],
    });
  });
  it('get user by name', async () => {
    const result = await testServer.executeOperation({
      query: userByNameQuery,
      variables: { name: 'Jakub' },
    });
    expect(result.data).toMatchSnapshot({
      getUserByName: {
        id: expect.any(String),
        name: expect.any(String),
        married: expect.any(Boolean),
        age: expect.any(Number),
      },
    });
  });
});
