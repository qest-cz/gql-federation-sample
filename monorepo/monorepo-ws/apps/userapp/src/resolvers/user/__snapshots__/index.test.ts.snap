// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user resolver get user by name 1`] = `
Object {
  "getUserByName": Object {
    "age": Any<Number>,
    "id": Any<String>,
    "married": Any<Boolean>,
    "name": Any<String>,
  },
}
`;

exports[`user resolver should return articles for user 1`] = `
Object {
  "users": Array [
    Object {
      "age": Any<Number>,
      "friends": Array [
        Object {
          "name": Any<String>,
        },
        Object {
          "name": Any<String>,
        },
      ],
      "id": Any<String>,
      "married": Any<Boolean>,
      "name": Any<String>,
    },
    Object {
      "age": Any<Number>,
      "friends": Array [],
      "id": Any<String>,
      "married": Any<Boolean>,
      "name": Any<String>,
    },
    Object {
      "age": Any<Number>,
      "friends": Array [],
      "id": Any<String>,
      "married": Any<Boolean>,
      "name": Any<String>,
    },
    Object {
      "age": Any<Number>,
      "friends": Array [],
      "id": Any<String>,
      "married": Any<Boolean>,
      "name": Any<String>,
    },
  ],
}
`;
